### ISSUE TITLE
Syntax error in function body produces error mark too early

### ISSUE THREAD

**Issue by rillig** [holgerbrandl/r4intellij#213]
~~~r
error.demo <- function(df)
{
  message('hello')
  df$$key <- 'value'
}
~~~

![image](https://user-images.githubusercontent.com/3233724/66685657-75077180-ec7d-11e9-8112-7d77714c2373.png)

There is a syntax error in the `df$$key` line.
In the IDE, the syntax error is marked at the `{`.
For long functions, this makes it impossible to spot the actual error.
The error marker should be at the point of the actual error, which in this case is the second `$`.

**Comment by holgerbrandl** [comment]
I fully agree it's an annoying bug. Unfortunately, my parser skills are to limited to fix this.
